dev/ci/scripts/utils/find_homegfs.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+"""
+Utility module to find the HOMEgfs (repository root) directory by traversing up
+the file system until finding a directory that contains the .github subdirectory.
+"""
+
+import os
+from pathlib import Path
+
+
+def find_homegfs(start_path=None):
+    """
+    Find the HOMEgfs directory by traversing up the file system until
+    finding a directory that contains the .github subdirectory.
+
+    Parameters
+    ----------
+    start_path : str or Path, optional
+        The path to start searching from. If not provided, the current
+        directory will be used.
+
+    Returns
+    -------
+    Path
+        The full path to the HOMEgfs directory.
+
+    Raises
+    ------
+    ValueError
+        If the HOMEgfs directory cannot be found.
+    """
+    # If start_path is not provided, use the directory of the calling script
+    if start_path is None:
+        # Get the path of the calling script
+        import inspect
+        frame = inspect.currentframe().f_back
+        start_path = os.path.dirname(os.path.abspath(frame.f_code.co_filename))
```

Comment:
```suggestion
        start_path = os.getcwd()
```
No need for inspect, assume starting from cwd.

Reply:
Agreed ... we need to par down the imports and up tic the readability out tipping the robustness of this choice.